flexmeasures/api/v1/tests/conftest.py

```diff
@@ -33,6 +34,7 @@ def setup_api_test_data(db, setup_roles_users, add_market_prices):
     # Create 1 test asset for the anonymous user
     test_asset_type = AssetType(name="test-type")
     db.session.add(test_asset_type)
+    db.session.add(GenericAssetType(name="test-type"))
```

Comment:
I see you're maintaining for all test data that there is a `GenericAssetType` for each `AssetType` before the `Asset` is made, so that the creation of the corresponding `Sensor` succeeds.
Wouldn't it be simpler (in this PR, but also for later refactoring) to enforce the existence of a `GenericAssetType` in `AssetType.__init__`, similar to how we enforce a `Sensor` for each `Asset` at the moment?